src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -841,8 +841,17 @@ HRESULT CProfilerManager::SetEventMask(DWORD dwEventMask)
     }
     else
     {
-        CLogging::LogError(_T("SetEventMask can only be called during initialize of Host or InstrumentationMethod"));
-        return E_FAIL;
+        // Prevent immutable flags from being set.
+        if ((dwEventMask & COR_PRF_MONITOR_IMMUTABLE) !=
```

Comment:
I think you need to perform a similar check for COR_PRF_HIGH_MONITOR_IMMUTABLE in SetEventMask2, but I don't know if you need to absolutely fix that right now.

Reply:
Yep, I saw that as well. I'll try to address that next

---
In reply to: [380925449](https://github.com/microsoft/CLRInstrumentationEngine/pull/202#discussion_r380925449) [](ancestors = 380925449)